source/core/module_controlEvents.js

```diff
@@ -0,0 +1,78 @@
+// Using a modified UMD module format. Specifically a modified returnExports (no dependencies) version
+(function (root, globalName, factory) {
+    'use strict';
+    var buildGlobalNamespace = function () {
+        var buildArgs = Array.prototype.slice.call(arguments);
+        return globalName.split('.').reduce(function (currObj, subNamespace, currentIndex, globalNameParts) {
+            var nextValue = currentIndex === globalNameParts.length - 1 ? factory.apply(undefined, buildArgs) : {};
+            return currObj[subNamespace] === undefined ? (currObj[subNamespace] = nextValue) : currObj[subNamespace];
+        }, root);
+    };
+
+    if (typeof define === 'function' && define.amd) {
+        // AMD. Register as a named module.
+        define(globalName, [], factory);
+    } else if (typeof module === 'object' && module.exports) {
+        // Node. "CommonJS-like" for environments like Node but not strict CommonJS
+        module.exports = factory();
+    } else {
+        // Browser globals (root is window)
+        buildGlobalNamespace();
+    }
+}(this, 'NationalInstruments.Vireo.ModuleBuilders.assignControlEvents', function () {
+    'use strict';
+
+    /* global Map */
+
+    // Static Private Variables (all vireo instances)
+
+    var assignControlEvents = function (Module, publicAPI) {
+        Module.controlEvents = {};
+        publicAPI.controlEvents = {};
+
+        var registerForControlEvent = function () {
+            // Dummy no-op function.
```

Comment:
Maybe these functions should throw and say an implementation is needed for event handling

Reply:
Sounds reasonable to me.
It can also provide errors for someone registering the functions late. ie creating a vireo instance, calling via load, registering functions, calling execute and wondering why the functions never fire.

Reply:
Ok, done.

Reply:
I'm getting an 'unexpected empty function' from eslint.  Can you tell me how to fix?
(Also, I can't run eslint locally; npm run lint hangs indefinitely for me and I don't know how to debug, so I have to rely on CI, so making any JS changes is really tedious for me.)

Reply:
You can try running `npm install` locally in-case your dependencies are out of date.
There are also eslint plugins for vscode and visual studio that are pretty helpful during development.

You can drop a comment in a function to describe why it is being left blank like the `// Dummy no-op function.` above.

Reply:
Turns out it was taking forever not because it was misconfigured but because I had a webvi saved in my test-it directory I had been debugging at some point, and it had a ni-webvi-resource-v0 folder containing a large number of js files.

I think make rules should NEVER use wildcards to descend folders but always refer to explicit file lists, so  'eslint .' should be reworked.